### PR TITLE
The meal db

### PR DESCRIPTION
**Pull request by FilipM79**
Implementing a local database

### CODE REVIEW

File: app/src/main/java/com/example/foody/shared/data/LocalDataSource.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import com.example.foody.shared.data.database.RecipesDao
+import com.example.foody.shared.data.database.RecipesEntity
+import kotlinx.coroutines.flow.Flow
+import javax.inject.Inject
+import javax.inject.Singleton
+
+// this is a source for local Room database
+@Singleton
+class LocalDataSource @Inject constructor(
```
Comment (vlad-at-heetch) [inline app/src/main/java/com/example/foody/shared/data/LocalDataSource.kt:11]: What is the purpose of this class.

Comment (FilipM79) [inline app/src/main/java/com/example/foody/shared/data/LocalDataSource.kt:11]: Implementing assisted injection ...

File: app/src/main/java/com/example/foody/recipe_search/presentation_mvvm/ui/SearchScreen.kt [hunk 2]
```diff
@@ -199,10 +235,10 @@ private fun RecipeImage(imageUrl: String) {
 @Composable
 @Preview
 private fun Preview() {
-    SearchScreenContent(
-        state = SearchScreenState(searchTerm = "", recipes = emptyList()),
-        performSearch = { Unit },
-        navigateWith = { Unit },
-        onValueChanged = { Unit },
-    )
+//    SearchScreenContent(
```
Comment (Vladimir-Markovic) [inline app/src/main/java/com/example/foody/recipe_search/presentation_mvvm/ui/SearchScreen.kt]: Uncomment

**Review by Vladimir-Markovic — APPROVED** [review]
👏 🚀

**Pull request merged**